src/lib/pages/frontsite/PatronProfile/PatronCurrentLoans/BrwReqLoanExtendButton.js

```diff
@@ -0,0 +1,215 @@
+import { borrowingRequestApi } from '@api/ill';
+import { withCancel } from '@api/utils';
+import { Error } from '@components/Error';
+import { invenioConfig } from '@config';
+import _get from 'lodash/get';
+import _isEmpty from 'lodash/isEmpty';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { Button, Icon, Popup } from 'semantic-ui-react';
+
+const CONFIG_EXPIRE_DAYS = invenioConfig.circulation.loanWillExpireDays;
+
+const INFO_MESSAGES = {
+  SUCCESS: documentTitle =>
+    `The loan extension for "${documentTitle}" has been requested to the library and you should receive an answer soon.`,
+  IS_TOO_EARLY: reqDate =>
+    `It is too early for extending the loan. You can request an extension from
+${reqDate
+  .minus({ days: CONFIG_EXPIRE_DAYS })
+  .toLocaleString({ month: 'long', day: 'numeric' })}.`,
+  MAX_EXTENSIONS:
+    'You have reached the max number of extensions for this loan!',
+};
+
+const REQUEST_MESSAGES = {
+  IS_PENDING: requestDate =>
+    `Extension pending since ${requestDate.toLocaleString({
+      month: 'long',
+      day: 'numeric',
+    })}`,
+  IS_DECLINED: 'Extension declined by the library',
+};
+
+class BorrowingRequestFetcher {
+  get = async pid => {
+    this.cancellablePromise = withCancel(borrowingRequestApi.get(pid));
+    const resp = await this.cancellablePromise.promise;
+    return resp.data;
+  };
+
+  cancel = () => {
+    this.cancellablePromise && this.cancellablePromise.cancel();
+  };
+}
+
+class ExtensionRequester {
+  doRequest = async brwReqPid => {
+    this.cancellablePromise = withCancel(
+      borrowingRequestApi.requestExtension(brwReqPid)
+    );
+    const resp = await this.cancellablePromise.promise;
+    return resp.data;
+  };
+
+  cancel = () => {
+    this.cancellablePromise && this.cancellablePromise.cancel();
+  };
+}
+
+const validateLoanExtension = brwReqLoan => {
+  let isValid = true;
+  let msg = null;
+
+  const isTooEarly =
+    brwReqLoan.metadata.end_date.diffNow('days').days > CONFIG_EXPIRE_DAYS;
```

Comment:
IMO this should come from the backend, but maybe good enough for now

Reply:
Yes, it makes sense since the backend will fail otherwise. I can add it to the serializers I guess! Do you want me to change it?

Reply:
maybe later, I will create an issue for this (good-first-time I would say)